packages/relay/src/lib/eth.ts

```diff
@@ -1106,57 +1107,66 @@ export class EthImpl implements Eth {
     return nonceCount;
   }
 
-  /**
-   * Submits a transaction to the network for execution.
-   *
-   * @param transaction
-   */
-  async sendRawTransaction(transaction: string, requestId?: string): Promise<string | JsonRpcError> {
+  async parseRawTxAndPrecheck(transaction: string, requestId?: string){
     const requestIdPrefix = formatRequestIdMessage(requestId);
-    this.counter.labels(EthImpl.ethSendRawTransaction, transaction.substring(0,10)).inc();
-
     let interactingEntity = '';
     let originatingAddress = '';
+    let parsedTx: EthersTransaction;
     try {
-      const parsedTx = Precheck.parseTxIfNeeded(transaction);
-      interactingEntity = parsedTx.to ? parsedTx.to.toString() : '';
-      originatingAddress = parsedTx.from ? parsedTx.from.toString() : '';
+      parsedTx = Precheck.parseTxIfNeeded(transaction);
+      interactingEntity = parsedTx.to?.toString() || '';
+      originatingAddress = parsedTx.from?.toString() || '';
       this.logger.trace(`${requestIdPrefix} sendRawTransaction(from=${originatingAddress}, to=${interactingEntity}, transaction=${transaction})`);
 
       const gasPrice = Number(await this.gasPrice(requestId));
       await this.precheck.sendRawTransactionCheck(parsedTx, gasPrice, requestId);
+      return parsedTx;
     } catch (e: any) {
       this.logger.warn(`${requestIdPrefix} Error on precheck sendRawTransaction(from=${originatingAddress}, to=${interactingEntity}, transaction=${transaction})`);
       throw this.genericErrorHandler(e);
     }
+  }
+
+  /**
+   * Submits a transaction to the network for execution.
+   *
+   * @param transaction
+   */
+  async sendRawTransaction(transaction: string, requestId?: string): Promise<string | JsonRpcError> {
+    const requestIdPrefix = formatRequestIdMessage(requestId);
+    this.counter.labels(EthImpl.ethSendRawTransaction, transaction.substring(0,10)).inc();
 
+    const parsedTx = await this.parseRawTxAndPrecheck(transaction, requestId);
     const transactionBuffer = Buffer.from(EthImpl.prune0x(transaction), 'hex');
+    const computedHash = EthImpl.prepend0x(createHash('keccak256').update(transactionBuffer).digest('hex'));
```

Comment:
Move this to the last catch since that's the only place it seems to be used. We should avoid calculating this if there are flows that won't require the calculation